### ISSUE TITLE
Building with podman (when Docker is not installed) is not working

### ISSUE THREAD

**Issue by mykaul** [noobaa/noobaa-core#5995]
There's something wrong with the Makefile, just did not nail it down yet.
1. Line is wrong, as it assumes Docker is installed.
```
NAME_POSTFIX?="$(shell docker ps -a | wc -l | xargs)"
```

2. Regardless, it fails (the 1st error is from the above line):
```
[ykaul@ykaul noobaa-core]$ make all
/bin/sh: docker: command not found
\n  Error: You must have container engine installed\n
make: *** [Makefile:28: assert-container-engine] Error 1
[ykaul@ykaul noobaa-core]$ podman version
Version:            1.8.2
RemoteAPI Version:  1
Go Version:         go1.14
OS/Arch:            linux/amd64
```

running from a37314640c7968489ab64d2f48206207921eaffb on Fedora 32, with podman but without docker installed.

**Comment by mykaul** [comment]
This fixes it for me, but I'm not sure is the right fix:
```
[ykaul@ykaul noobaa-core]$ git diff
diff --git a/Makefile b/Makefile
index 71dc7f25c..104986326 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,10 @@
-CONTAINER_ENGINE?=$(shell docker version >/dev/null 2>&1 && echo docker)
+CONTAINER_ENGINE?="$(shell docker version >/dev/null 2>&1 && echo docker)"
 ifeq ($(CONTAINER_ENGINE), "")
-       CONTAINER_ENGINE==$(shell podman version >/dev/null 2>&1 && echo podman)
+       CONTAINER_ENGINE=$(shell podman version >/dev/null 2>&1 && echo podman)
 endif
 
 GIT_COMMIT?="$(shell git rev-parse HEAD | head -c 7)"
-NAME_POSTFIX?="$(shell docker ps -a | wc -l | xargs)"
+NAME_POSTFIX?="$(shell ${CONTAINER_ENGINE} ps -a | wc -l | xargs)"
 BUILDER_TAG?="noobaa-builder"
 TESTER_TAG?="noobaa-tester"
 NOOBAA_TAG?="noobaa"
```

**Comment by liranmauda** [comment]
@mykaul 
It looks like the correct fix, can you open a PR?

**Comment by liranmauda** [comment]
about:
`NAME_POSTFIX?="$(shell docker ps -a | wc -l | xargs)"`
it should be replaced with:
`NAME_POSTFIX?="$(shell $(CONTAINER_ENGINE) ps -a | wc -l | xargs)"`

**Comment by mykaul** [comment]
> @mykaul
> It looks like the correct fix, can you open a PR?

Sure, once I validate the fix. Still failing on 'Error: EMFILE: too many open files, open '/noobaa/frontend/package.json' issues.

**Comment by liranmauda** [comment]
> > @mykaul
> > It looks like the correct fix, can you open a PR?
> 
> Sure, once I validate the fix. Still failing on 'Error: EMFILE: too many open files, open '/noobaa/frontend/package.json' issues.

Are you referring to #5858?
It usually depends on each different host which has a different limit to the number of open files.

It might be that the frontend has a lot of dependencies, so @nb-ohad might know

**Comment by liranmauda** [comment]
Closing as it was fixed by #5996
